pkg/engine/apicall/apiCall.go

```diff
@@ -162,6 +162,17 @@ func (a *apiCall) buildHTTPRequest(ctx context.Context, apiCall *kyvernov1.Conte
 		if token != "" && req != nil {
 			req.Header.Add("Authorization", "Bearer "+token)
 		}
+
+		if apiCall.Headers != nil && req != nil {
```

Comment:
Since the logic in the deferred has grow, I suggest refactoring the method a bit to be easier to maintain. Something like this?

```go
func (a *apiCall) buildHTTPRequest(ctx context.Context, apiCall *kyvernov1.ContextAPICall) (req *http.Request, err error) {
	if apiCall.Service == nil {
		return nil, fmt.Errorf("missing service")
	}

	if apiCall.Method == "GET" {
		req, err = http.NewRequestWithContext(ctx, "GET", apiCall.Service.URL, nil)
	} else if apiCall.Method == "POST" {
		data, dataErr := a.buildRequestData(apiCall.Data)
		if dataErr != nil {
			return nil, dataErr
		}

		req, err = http.NewRequest("POST", apiCall.Service.URL, data)
	} else {
		return nil, fmt.Errorf("invalid request type %s for APICall %s", apiCall.Method, a.entry.Name)
	}

	if err != nil {
		return nil, err 
	}

	token := a.getToken()
	if token != ""  {
		req.Header.Add("Authorization", "Bearer "+token)
	}

	if apiCall.Headers != nil {
		for _, header := range apiCall.Headers {
			if header.Key == "Authorization" {
				req.Header.Set(header.Key, header.Value)
			} else {
				// TODO: Pull values if they are secrets
				req.Header.Add(header.Key, header.Value)
			}
		}
	}

	return req, nil 
}
```